node/src/braid/mod.rs

```diff
@@ -248,7 +254,70 @@ impl Braid {
             }
         }
     }
+    /// utility function for GetBeadsAfter request
+    pub fn get_beads_after(&self, old_tips: Vec<BeadHash>) -> Option<Vec<Bead>> {
+        let old_tips: HashSet<BeadHash> = old_tips.into_iter().collect();
+        tracing::warn!(
+            old_tips=?old_tips,"Tips received from the peer for which beads are requested for during IBD"
+        );
+        //In case no tips are present i.e. the new braid-node has been initialized
+        if old_tips.len() == 0 {
+            return Some(self.beads.clone());
+        }
+        let mut response_beads = Vec::new();
+        let mut found_start = false;
+        let mut smallest_index = usize::MAX;
+        //finding the starting index
+        for hash in &old_tips {
+            if let Some(&index) = self.bead_index_mapping.get(hash) {
+                if index < smallest_index {
+                    smallest_index = index;
+                }
+                found_start = true;
```

Comment:
found_start doesn't do anything and isn't needed. If we error above on missing hashes, we should probably also error if old_tips is empty. Then after that smallest_index will always be the smallest in the following code without the found_start variable.